File: MediCareWebSpa/src/app/DTOs/models/medicament.ts
import { MedicamentType } from "../../enums/medicament-type";

export class Medicament {
  id: number;
  name: string;
  description: string;
  medicamentType: MedicamentType;
  prescriptionRequired: boolean;

  constructor(
    id: number,
    name: string,
    description: string,
    medicamentType: MedicamentType,
    prescriptionRequired: boolean
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.medicamentType = medicamentType;
    this.prescriptionRequired = prescriptionRequired;
  }
}

File: MediCareWebApi/Models/Medicament.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.Models
{
	public class Medicament : Entity
	{
		[Required]
		[MaxLength(256)]
		public string Name { get; set; }

		[Required]
		[MaxLength(256)]
		public string Description { get; set; }

		[Required]
		public MedicamentType MedicamentType { get; set; }

		[Required]
		public bool PrescriptionRequired { get; set; }

		public ICollection<PrescriptionMedicament> MedicamentPrescriptions { get; set; } = new List<PrescriptionMedicament>();
	}
}
